File: v1/provider/ollama.py
"""
Ollama Provider —— 对接 Ollama 本地推理（OpenAI 兼容 API）。

默认端点: http://localhost:11434/v1
"""
from typing import Dict

from provider.base import BaseProvider


class OllamaProvider(BaseProvider):
    """Ollama 本地推理 Provider"""

    @property
    def provider_name(self) -> str:
        return "ollama"

    def _build_url(self) -> str:
        return f"{self.base_url}/chat/completions"

    def _build_headers(self) -> Dict[str, str]:
        headers = {"Content-Type": "application/json"}
        # Ollama 通常不需要 API Key，但如果配置了则带上
        if self.api_key and self.api_key != "not-needed":
            headers["Authorization"] = f"Bearer {self.api_key}"
        return headers

File: provider/ollama.go
package provider

import "fmt"

type OllamaProvider struct {
	BaseURL       string
	APIKey        string
	ModelName     string
	MaxTokens     int
	ContextWindow int
	Temperature   float64
	TopP          float64
}

func NewOllamaProvider(baseURL, apiKey, modelName string, maxTokens, contextWindow int, temperature, topP float64) *OllamaProvider {
	return &OllamaProvider{
		BaseURL:       baseURL,
		APIKey:        apiKey,
		ModelName:     modelName,
		MaxTokens:     maxTokens,
		ContextWindow: contextWindow,
		Temperature:   temperature,
		TopP:          topP,
	}
}

func (p *OllamaProvider) ProviderName() string { return "ollama" }

func (p *OllamaProvider) BuildURL() string {
	return fmt.Sprintf("%s/chat/completions", p.BaseURL)
}

func (p *OllamaProvider) BuildHeaders() map[string]string {
	h := map[string]string{"Content-Type": "application/json"}
	if p.APIKey != "" && p.APIKey != "not-needed" {
		h["Authorization"] = "Bearer " + p.APIKey
	}
	return h
}

func (p *OllamaProvider) GetContextWindow() int { return p.ContextWindow }
func (p *OllamaProvider) GetMaxTokens() int     { return p.MaxTokens }
func (p *OllamaProvider) GetMaxInput() int      { return p.ContextWindow - p.MaxTokens - 2048 }
